cli/tests/activate.bats

```diff
@@ -934,6 +1610,24 @@ EOF
   assert_line "/custom/zdotdir"
 }
 
+# bats test_tags=activate,activate:custom_zdotdir,activate:custom_zdotdir:fish
+@test "fish: preserve custom ZDOTDIR" {
+  "$FLOX_BIN" delete -f
+  "$FLOX_BIN" init
+  FLOX_SHELL=fish ZDOTDIR=/custom/zdotdir run "$FLOX_BIN" activate -- echo '$ZDOTDIR'
+  assert_success
+  assert_line "/custom/zdotdir"
+}
+
+# bats test_tags=activate,activate:custom_zdotdir,activate:custom_zdotdir:tcsh
+@test "tcsh: preserve custom ZDOTDIR" {
+  "$FLOX_BIN" delete -f
+  "$FLOX_BIN" init
+  FLOX_SHELL=tcsh ZDOTDIR=/custom/zdotdir run "$FLOX_BIN" activate -- echo '$ZDOTDIR'
+  assert_success
+  assert_line "/custom/zdotdir"
+}
+
```

Comment:
That was a lot of tests to port! Thanks!

Reply:
Thanks, a lot of copy & paste.